internal/git/gitsync_processor.go

```diff
@@ -434,7 +458,7 @@ func getLatestCommitHash(repo *git.Repository, refName string) (*plumbing.Hash,
 	return commitHash, err
 }
 
-func NewGitSyncProcessor(ctx context.Context, gitSync *v1alpha1.GitSync, kubeClient kubernetes.Client, clusterName string) (*GitSyncProcessor, error) {
+func NewGitSyncProcessor(ctx context.Context, gitSync *v1alpha1.GitSync, kubeClient kubernetes.Client, clusterName string, repoCred map[string]*controllerconfig.GitCredential) (*GitSyncProcessor, error) {
```

Comment:
Sorry, I know my original thought was to just pass in the credentials map here, but now I'm thinking that either we pass in the `configManager`, or maybe even better we could just have a singleton ConfigManager that anyone can access...